wasmi_v1/src/arena/mod.rs

```diff
@@ -162,7 +162,7 @@ where
 #[derive(Debug)]
 pub struct Iter<'a, Idx, T> {
     iter: iter::Enumerate<slice::Iter<'a, T>>,
-    __marker: PhantomData<fn() -> Idx>,
+    __marker: PhantomData<Idx>,
```

Comment:
This change would change semantic in a way that is not useful for a non-owning `Iter`. Please revert. Read more here: https://doc.rust-lang.org/nomicon/phantom-data.html

Reply:
This change request is still outstanding before we can merge. Rest looks good to me.